### ISSUE TITLE
`document.getElementById` issues

### ISSUE THREAD

**Issue by maxshuty** [Sarah-Workman/Vue-stopwatch-#1]
For this code you're doing component logic in the store. Anytime you find yourself writing `document.` in Vue then you're probably doing it wrong, but you don't know what you don't know :)
```		
createP(state) {
  state.lapTime = `${state.outputhours}:${state.outputminutes}:${state.outputseconds}`;
  let lapContainer = document.getElementById("lapContainer");
  const para = document.createElement("p");
  para.innerText = state.lapTime;
  state.laps = Array.from(
    document.getElementById("lapContainer").childNodes
    ).map((node) => node.textContent);
  let lastItem = state.laps[state.laps.length - 1];
  let currentItem = state.lapTime;
  
  if (currentItem !== lastItem) {
    lapContainer.appendChild(para);
  }
},
```
 Instead of this code above, your store can contain an array of objects with lap data like this (ignore the values, just dummy data here):
 
 ```
state.laps = 
 [
   { time: "1", startTime: "1", endTime: "1", ...},
   { time: "2", startTime: "2", endTime: "2", ...},
   { time: "3", startTime: "3", endTime: "3", ...},
   { time: "4", startTime: "4", endTime: "4", ...},
   { time: "5", startTime: "5", endTime: "5", ...},
   ...
 ]
```
(Note: don't set your state like that above, use a mutation to push a new lap to the array, I'm just showing you for example what the `state.laps` would look like.)

 And then inside of your component where you want to create a new `<p>` element you can `mapState` and use this object above.  In Vue you would do something like this:
 
 ```
 <p v-for="(lap, index) in laps" :key="index">
  Time: {{ lap.time }} - Start Time: {{ lap.startTime }} - End Time: {{ lap.endTime }}
</p>
 ```
 
Now you can get rid of `document.getElementById` and all of that stuff in your store and it's nice, clean, and easily readable right inside of the component. If you are like "well what if I want to reuse this lap information all over the place, do I really need to type out that `<p v-for...` every single time? The answer is **no** - simply create a component does this and reuse _that_ component in one or many places.

The above example would output this:

```
<p>Time: 1 - Start Time: 1 - End Time: 1</p>
<p>Time: 2 - Start Time: 2 - End Time: 2</p>
<p>Time: 3 - Start Time: 3 - End Time: 3</p>
<p>Time: 4 - Start Time: 4 - End Time: 4</p>
<p>Time: 5 - Start Time: 5 - End Time: 5</p>
```

**Comment by Sarah-Workman** [comment]
I thought that using `getElementById("#element")` defeated the purpose of using a state management, but wasn't sure how to approach programmatically adding the ps within the store without it. So now I know! :) 

I set my store up like this.
```		laps: [
			{
				lapHour: undefined,
				lapMinute: undefined,
				lapSecond: undefined,
			},
		],
```
The steps I took to fix this issue was firstly I deconstructed the array of objects,  laps, and created three separate variables named lapHour, lapMinute, and lapSecond.  I then assigned those variables to what the timer output should be with the placeholders. I put all of my variables into an interpolation in the stopwatch format then stored them in object lapTime.
This captured the current time that is displayed in the stopwatch rectangle so that I could manipulate the object later. 
```			var [{ lapHour, lapMinute, lapSecond }] = state.laps;
			lapHour = state.outputhours;
			lapMinute = state.outputminutes;
			lapSecond = state.outputseconds;
			state.lapTime = `${lapHour}:${lapMinute}:${lapSecond}`; 
```

Secondly,  I wanted to make sure that the laps aren't duplicate so I kept the lastItem and currentItem logic but updated it to component logic for the store. Last item will be based off the last lap in the array using `state.laps.length-1`, and that will be compared to the current item that uses the incoming array object. Once that check has passed it should push lapTime to the laps object array in the store, which is then displayed in the lap component as a v-for loop creating individual array objects called lap. 
```		let lastItem = state.laps[state.laps.length - 1];
			let currentItem = state.lapTime;

			if (currentItem !== lastItem) {
				state.laps.push(state.lapTime);
				console.log(state.lapTime);
				console.log(state.laps);
			} 
```
For the reset laps button I kept the while loop and set my logic as long as the array laps has a length of greater than 1 it will remove the last item in the array. 

```	clearLaps(state) {
			while (state.laps.length > 1) {
				state.laps.pop();
			}
		}, 
```


The issue I'm running into now is that it displays the object curly braces as the first object in the array. 
![image](https://user-images.githubusercontent.com/55566625/189708572-d3fc95b5-67bd-417e-9f50-2d2c534bd919.png)
So say I set the clearLaps while condition to run while `while laps.length > 0 ` it removed the curly braces and creates an error when trying to start laps again. 

![image](https://user-images.githubusercontent.com/55566625/189710692-5fe6d412-81a8-4adc-b4e6-23fefd6b62d9.png)
